include/revng/Model/Binary.h

```diff
@@ -6,593 +6,16 @@
 
 #include "llvm/ADT/SmallString.h"
 
-#include "revng/ADT/KeyedObjectContainer.h"
-#include "revng/ADT/MutableSet.h"
-#include "revng/ADT/SortedVector.h"
-#include "revng/ADT/UpcastablePointer.h"
-#include "revng/ADT/UpcastablePointer/YAMLTraits.h"
-#include "revng/Model/Register.h"
-#include "revng/Model/TupleTree.h"
-#include "revng/Model/Type.h"
-#include "revng/Support/MetaAddress.h"
-#include "revng/Support/MetaAddress/YAMLTraits.h"
-#include "revng/Support/YAMLTraits.h"
-
-// Forward declarations
-namespace model {
-class VerifyHelper;
-class Function;
-class DynamicFunction;
-class Binary;
-class FunctionEdge;
-class CallEdge;
-class BasicBlock;
-class Segment;
-} // namespace model
+#include "revng/Model/GeneratedBinary.h"
+#include "revng/Model/GeneratedPrimitiveTypeKind.h"
```

Comment:
Why do we need this? Can't we generate this include in `revng/Model/GeneratedBinary.h`?

Reply:
Yep, there's no need to include `PrimitiveTypeKind.h`, I dropped it.